### DISCUSSION TITLE
Adding method for layout

### DISCUSSION THREAD

**Discussion by evenreven** [yippee-fun/phlex#240] [Help] [answered]
Loving this project so far, so to get my feet wet I'm trying to recreate a hobby project's view directory with Phlex. I wanted to see if I could do it without erb at all, so I deleted all the erb files and tried the layout generator, and I'm now rendering a generated page (using the layout) directly from the controller. Works really well!

Apart from one thing: I'm using vite_rails and need `vite_javascript_tag`, `vite_client_tag` and `vite_stylesheet_tag` available to my views. Can I register more helpers myself? I found where they were defined, and tried to add more methods through adding a similar module structure, but it failed with unitialized constant. I'm not a gem developer, and have very little experience with the innards of the lib directory, so I'm probably doing something wrong. It's probably out of scope to add the vite_rails helpers to the Phlex code?

**Comment by joeldrapper** [comment, marked as answer]
Hey @evenreven, you can definitely define more helpers yourself. Helpers are just instance methods. If you only have one layout view, you may just want to define a private method directly on that Layout view.

```ruby
private

def vite_javascript_tag
  raw helpers.vite_javascript_tag
end
```

If you need the tag to be available in other places, you could either define it on a module and include that module in both places, or you could make an abstract `ApplicationLayout` view and have both all layouts inherit from it.

**Reply by evenreven** [reply]
Today I learned! Thanks so much, @joeldrapper, tried it now and it works well. I'm fine with the private method at least for now, I'll make an abstraction for it later if I need it. Thanks again!

**Reply by joeldrapper** [reply]
🙌 anything that you can write in the template method, you can write in any instance method called from the template and it works the same way. It's a great way to refactor views and make them easier to understand.
